File: julia/bob/bob.jl
function is_question(msg::String)::Bool
    return strip(msg)[end] == '?'
end

function is_yell(msg::String)::Bool
    return msg == uppercase(msg) && any(isuppercase, msg)
end

function bob(stimulus)
    if strip(stimulus) == ""
        return "Fine. Be that way!"
    elseif is_yell(stimulus) && is_question(stimulus)
        return "Calm down, I know what I'm doing!"
    elseif is_yell(stimulus)
        return "Whoa, chill out!"
    elseif is_question(stimulus)
        return "Sure."
    else
        return "Whatever."
    end
end

File: python/bob/bob.py
def is_question(msg: str) -> bool:
    return msg.strip().endswith("?")


def is_yell(msg: str) -> bool:
    return msg.upper() == msg and msg.lower() != msg


def is_silence(msg: str) -> bool:
    return len(msg.strip()) == 0


def response(hey_bob: str) -> str:
    if is_question(hey_bob) and is_yell(hey_bob):
        return "Calm down, I know what I'm doing!"

    if is_silence(hey_bob):
        return "Fine. Be that way!"

    if is_question(hey_bob):
        return "Sure."

    if is_yell(hey_bob):
        return "Whoa, chill out!"

    return "Whatever."
